src/app/state/rest/useApiRequest.ts

```diff
@@ -45,55 +32,36 @@ const useApiRequest = <SCHEMA>({ url, group }: Config) => {
     return Promise.reject(details)
   }, [setError, addErrorFlashMessage])
 
-  /**
-   * Execute API request
-   */
+  // Execute an API request
   const exec = useCallback(async (method: Method, payload?: {}) => {
-    setIsBusy(true)
-
     try {
-      // collect cancelTokens, we want to be able to cancel all requests on unMount.
-      const cancelToken = axios.CancelToken.source()
-      cancelTokens.current.push(cancelToken)
-
-      const headers = {
-        Authorization: `Bearer ${ authorizationToken }`
-      }
-
-      const promise: Promise<AxiosResponse<SCHEMA>> =
-        // Is this request pending already?
-        pending[url] === undefined
-          // ...no its not! Execute request
-          ? api.request<SCHEMA>({
-              headers,
-              method,
-              url: slashSandwich([process.env.REACT_APP_GATEWAY, url]),
-              data: payload,
-              cache: { key: () => group },
-              cancelToken: cancelToken.token
+      setIsBusy(true)
+
+      const promise: Promise<AxiosResponse<SCHEMA>> = pending[url]
+        ?  pending[url]
```

Comment:
There should also be a check for the `method` here, otherwise, for example, a GET request could be turned into a POST when the URL's are the same.